defender/middleware.py

```diff
@@ -18,9 +17,10 @@ def __init__(self, *args, **kwargs):
             # `LoginView` class-based view
             try:
                 from django.contrib.auth.views import LoginView
-                watch_login_method = method_decorator(watch_login)
+                our_decorator = watch_login()
+                watch_login_method = method_decorator(our_decorator)
                 LoginView.dispatch = watch_login_method(LoginView.dispatch)
             except ImportError:  # Django < 1.11
-                auth_views.login = watch_login(auth_views.login)
+                auth_views.login = watch_login()(auth_views.login)
```

Comment:
That looks weird but it's correct. The old decorator couldn't accept arguments and had a deep of 2. I had to add one more level in order to accept arguments so the first thing the new decorator expects now are the arguments (or the defaults) while here a function was passed in first instance. Does it make sense?